NUnit3 support: download and finding console exe.

f00f opened the pull request:
It's just a first step, command line syntax seems to be different as well.

Commit:
start to handle NUnit v3

- the ZIP structure changed
- the download url changed
- the command line is in AnyCPU so useX86 is obsolete
- works remain like handling the aguments (they have changed)

Commit:
NUnit3 support: download and finding console exe.

gluck commented on src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:
```diff
@@ -39,7 +39,9 @@ class NUnit extends ConventionTask {
 
     File getNunitExec() {
         assert getNunitHome(), "You must install NUnit and set nunit.home property or NUNIT_HOME env variable"
-        File nunitExec = nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
```
x86 seems to be a console param instead on NUnit v3 (as they're using a separate daemon to actually run the tests)

f00f replied:
Added in 01d515ab21b32de26e586ae05641d3f592950f69

Commit:
[bugfix] Fixed crash in version check

Commit:
Improved compatibility with NUnit3.

f00f commented:
* Added support for renamed command line parameters.
* Fixed a bug which caused a crash and the test faiures.

Some parameters still don't work (e.g. -exclude, -include) and I'm not sure what to do about -noshadow. However, I don't use these, yet, so I'll leave this task to someone else.

florian-gandon commented on src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:
```diff
@@ -39,7 +41,9 @@ class NUnit extends ConventionTask {
     File getNunitExec() {
         assert getNunitHome(), "You must install NUnit and set nunit.home property or NUNIT_HOME env variable"
-        File nunitExec = nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
+        File nunitExec = getNunitVersion().startsWith("3.")
```
What about a property "isV3 in the task ?

f00f replied:
Yeah, I like how you solved that.
I had a similar thing, but got confused with all the implicitness of gradle/Groovy and finally decided to remove it again.
I might port your approach.

f00f replied:
Merged in bd44ad5625c0c7a3377e1311b698222a0d9296e9

reeflog commented on src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:
```diff
@@ -39,7 +41,9 @@ class NUnit extends ConventionTask {
-        File nunitExec = nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
+        File nunitExec = getNunitVersion().startsWith("3.")
+            ? nunitBinFile('nunit3-console.exe')
+            : nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
```
I would log a warning about the usage of useX86 when it's V3 as it's ignored.

florian-gandon commented on src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy:
```diff
@@ -52,7 +52,12 @@ class NUnitPlugin implements Plugin<Project> {
+        def nunitCacheDirForVersion = new File(nunitCacheDir, NUnitName)
+
+        // special handling for nunit3 flat zip file
+        def zipOutputDir = version.startsWith("3.") ? nunitCacheDirForVersion : nunitCacheDir;
```
The download is working ? The repo on github doesn't have the same name (nunit intead of nunitv2).

f00f replied:
Well, for sure I had to change the download URL, but I cannot find that change in the commits now. Strange...

f00f replied:
Fixed in bd44ad5625c0c7a3377e1311b698222a0d9296e9

Commit:
Merge branch 'handle-v3' of https://github.com/reeflog/gradle-nunit-plugin

Commit:
Changed nunit version in test to 3.0.0

reeflog commented:
LGTM
